file_store/src/error.rs

```diff
@@ -37,6 +37,8 @@ pub enum Error {
     DbError(#[from] sqlx::Error),
     #[error("channel send error")]
     SendError(#[from] tokio::sync::mpsc::error::SendError<()>),
+    #[error("External Error")]
+    ExternalError(#[from] Box<dyn std::error::Error + Send + Sync>),
```

Comment:
What _is_ an eternal error? maybe addd a comment here because it's not clear that it's basically an "Other" error

In addition can you help understand what errors you are wrapping? Sometimes it's better to be explicit about your error but I can't tell if that's what should be happening here?

Reply:
I added comments to the code. Proposed change is emulating the generic Error type for external traits implementations. More clear solution would be change of the traits definition but it may take a lot of code rework.
Wrapped Errors in my case: type conversions errors, sql errors,  wrong timestamps, etc.